### PR TITLE
add dockerfile

### PR DESCRIPTION
**Pull request by sanbuphy**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Help china users build docker more effectively.

## Modification

fix the nvidia-GPG key error and python -i image source.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docker/serve/Dockerfile-cn [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# - Avoid Public GPG key error
+# - https://github.com/NVIDIA/nvidia-docker/issues/1631
+
+RUN apt-key del 7fa2af80
+RUN apt-get update && apt-get install -y --no-install-recommends wget
+RUN wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-keyring_1.0-1_all.deb
+RUN dpkg -i cuda-keyring_1.0-1_all.deb
+# (Optional, use Mirror to speed up downloads)
+# RUN sed -i 's/http:\/\/archive.ubuntu.com\/ubuntu\//http:\/\/mirrors.aliyun.com\/ubuntu\//g' /etc/apt/sources.list
+
```
Comment (ZwwWayne) [inline docker/serve/Dockerfile-cn:27]:
```suggestion
# Avoid Public GPG key error
# https://github.com/NVIDIA/nvidia-docker/issues/1631
RUN rm /etc/apt/sources.list.d/cuda.list \
  && rm /etc/apt/sources.list.d/nvidia-ml.list \
  && apt-get update \
  && apt-get install -y wget \
  && rm -rf /var/lib/apt/lists/* \
  && apt-key del 7fa2af80 \
  && apt-get update && apt-get install -y --no-install-recommends wget \
  && wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-keyring_1.0-1_all.deb \
  && dpkg -i cuda-keyring_1.0-1_all.deb
# (Optional, use Mirror to speed up downloads)
# RUN sed -i 's/http:\/\/archive.ubuntu.com\/ubuntu\//http:\/\/mirrors.aliyun.com\/ubuntu\//g' /etc/apt/sources.list

```

Comment (ZwwWayne) [reply docker/serve/Dockerfile-cn:27]: Please consider the suggest change to reduce using RUN command so that the docker image can be potentially smaller.

Comment (sanbuphy) [reply docker/serve/Dockerfile-cn]: OK, I have marged those RUN steps.

**Comment by ZwwWayne** [comment]
Hi @sanbuphy ,
Thanks for your kind contribution. Would you like to fix the lint issue so we can merge this PR?
The details can be found by clicking the item of lint check as shown below
image

If the lint is fixed and the updated dockerfile is verified, we can merge this PR.

**Comment by sanbuphy** [comment]
> Hi @sanbuphy , Thanks for your kind contribution. Would you like to fix the lint issue so we can merge this PR? The details can be found by clicking the item of lint check as shown below image
> 
> If the lint is fixed and the updated dockerfile is verified, we can merge this PR.
 
Hello, I have solved all problem and i have already try to build the dockerfile on my computer.
image

**Comment by ZwwWayne** [comment]
Great ! Thanks @sanbuphy ,
I find that the other two files in serve_cn is unnecessary. Is there anyway we can delete these two files and only keep the updated Dockerfile with acceleration in China?

**Comment by sanbuphy** [comment]
> Great ! Thanks @sanbuphy , I find that the other two files in serve_cn is unnecessary. Is there anyway we can delete these two files and only keep the updated Dockerfile with acceleration in China?

Yes, get it . I think you are right because we just use these two files in `serve` and we don't need to create the same files in `serve_cn`.

**Pull request merged**